Dc jc/update mesh guide (#5600)
fishtown-analytics/docs.getdbt.com: website/docs/best-practices/how-we-mesh/mesh-3-structures.md

### Cycle detection

Like resource dependencies, project dependencies are acyclic, meaning they only move in one direction. This prevents `ref` cycles (or loops), which lead to issues with your data workflows. For example, if project B depends on project A, a new model in project A could not import and use a public model from project B. Refer to [Project dependencies](/docs/collaborate/govern/project-dependencies#how-to-write-cross-project-ref) for more information.

:::note 

> This requirement is likely to change in the future, so stay tuned for updates!

Reviewer: 🚀

- **DRY applies to underlying data, not just code.** Regardless of your strategy, you should not be sourcing the same rows and columns into multiple nodes. When working within a mesh pattern it becomes increasingly important that we don't duplicate logic or data.

<Lightbox src="/img/best-practices/how-we-mesh/combined_splits.png" title="A simplified dbt DAG with two dotted lines representing both a vertical and horizontal split." />

## Determine your git strategy

A multi-project architecture can exist in a single repo (monorepo) or as multiple projects, with each one being in their own repository (multi-repo).

- If you're a **smaller team** looking primarily to speed up and simplify development, a **monorepo** is likely the right choice, but can become unwieldy as the number of projects, models and contributors grow.
- If you’re a **larger team with multiple groups**, and need to decouple projects for security and enablement of different development styles and rhythms, a **multi-repo setup** is your best bet.

> ## Projects, splits, and teams

Reviewer: 👍 good spot for this

> Since the laumch of dbt Mesh, the most common pattern we've seen is one where **projects are 1:1 aligned to teams**, and each project is its own codebase in its own repository. This isn’t a hard-and-fast rule: Some organizations want multiple teams working out of a monorepo. Some teams own multiple domains that feel awkward to keep combined.

Reviewer 2: ```suggestion
Since the launch of dbt Mesh, the most common pattern we've seen is one where projects are 1:1 aligned to teams, and each project has its own codebase in its own repository. This isn’t a hard-and-fast rule: Some organizations want multiple teams working out of a single repo, and some teams own multiple domains that feel awkward to keep combined.
```

> Users may need to contribute models across multiple projects. This is fine! There will be some more friction to do this, versus the monorepo — but this is **useful** friction, especially if upstreaming a change from a “spoke” to a “hub.” This should be treated like making an API change, one that the other team will be living with for some time to come. You should be concerned if your teammates find they need to make a coordinated change across multiple projects very frequently (every week), or as a key prerequisite for ~20%+ of their work.

Reviewer 2: ```suggestion
Users may need to contribute models across multiple projects and this is fine. There will be some friction doing this, versus a single repo, but this is _useful_ friction, especially if upstreaming a change from a “spoke” to a “hub.” This should be treated like making an API change, one that the other team will be living with for some time to come. You should be concerned if your teammates find they need to make a coordinated change across multiple projects very frequently (every week), or as a key prerequisite for ~20%+ of their work.
```

### Tips and tricks

> In the next page, we'll go through a more in depth example of how to split a monolithic project into multiple projects. Here are some tips to get you started when starting to think about using the splitting methods listed above on your own projects:

Reviewer 2: ```suggestion
The [implementation](/best-practices/how-we-mesh/mesh-4-implementation) page provides more in-depth examples of how to split a monolithic project into multiple projects. Here are some tips to get you started when considering the splitting methods listed above on your own projects:
```

> 1. Start by drawing a diagram of your teams doing data work. Map each team to a single dbt project. If you already have an existing monolithic project, and you’re onboarding *net-new teams,* this could be as simple as declaring the existing project as your “Hub” and creating new “Spoke” sandbox projects for each of those teams.

Reviewer 2: ```suggestion
1. Start by drawing a diagram of your teams doing data work. Map each team to a single dbt project. If you already have an existing monolithic project, and you’re onboarding _net-new teams,_ this could be as simple as declaring the existing project as your “hub” and creating new “spoke” sandbox projects for each team.
```

> 2. **Split off common foundations** when ****you know that multiple downstream teams will require the same data source. Those could be upstreamed into a centralized Hub, or split off into a separate foundational project. need some splits to facilitate other splits, e.g. source staging models in A that are used in both B + C (lack of project cycles).

Reviewer 2: ```suggestion
2. Split off common foundations when you know that multiple downstream teams will require the same data source. Those could be upstreamed into a centralized hub or split off into a separate foundational project. need some splits to facilitate other splits, for example, source staging models in A that are used in both B and C (lack of project cycles).
```

> 3. **Split again** to ~~introduce intentional friction /~~ encapsulate a particular set of models (e.g. for external export).

Reviewer 2: ```suggestion
3. Split again to introduce intentional friction and encapsulate a particular set of models (for example, for external export).
```

> 4. **Recombine** if you have “hot path” subsets of the DAG that you need to deploy with low latency, because it powers in-app reporting or operational analytics. It might make sense to have a different dedicated team own these data models (see principle 1), similar to how software services with significantly different performance characteristics often warrant dedicated infrastructure, architecture, and staffing.

Reviewer 2: ```suggestion
4. Recombine if you have “hot path” subsets of the DAG that you need to deploy with low latency because it powers in-app reporting or operational analytics. It might make sense to have a different dedicated team own these data models (see principle 1), similar to how software services with significantly different performance characteristics often warrant dedicated infrastructure, architecture, and staffing.
```
